frank/utilities.py

```diff
@@ -272,7 +321,77 @@ def cut_data_by_baseline(u, v, vis, weights, cut_range):
 
     return u_cut, v_cut, vis_cut, weights_cut
 
+def estimate_weights(u, v, V, nbins=300, log=True, use_median=False):
+    r"""
+    Estimate the weights using the variance of the binned visibilities. 
+
+    The estimation is done assuming that the variation in each bin is dominated
+    by the noise. This will be true if:
+        1) The source is axi-symmetric,
+        2) The uv-points have been deprojected,
+    Otherwise the variance may be dominated by real variations in the 
+    visibilities.
+
+    Parameters
+    ----------
+    u, v : array, unit = :math:`\lambda`
+        u and v coordinates of observations (deprojected).
+    V : array, unit = Jy
+        Obsersed visibility. If complex, the weights will be computed from the
+        average of the variance of the real and imaginary components, as in
+        CASA's statwt. Otherwise the variance of the real part is used.
```

Comment:
Can you add the link to statwt, https://casa.nrao.edu/docs/taskref/statwt-task.html?

Reply:
I'd rather not, since that page might change